From c650b6344f0e4b2701842f0e7f197f45aeb7a7ae Mon Sep 17 00:00:00 2001
From: christos <christos>
Date: Fri, 7 Dec 2012 20:54:39 +0000
Subject: [PATCH 1/1] block SIGINT while running children from shell scripts
 (From Roman Kollar) Other shells don't die when you ^C a script
 "#!shell\npython\n"

---
 sh.proc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sh.proc.c b/sh.proc.c
index 0c5fc25..2662d50 100644
--- a/sh.proc.c
+++ b/sh.proc.c
@@ -507,6 +507,7 @@ pjwait(struct process *pp)
     cleanup_push(&oset, sigprocmask_cleanup);
     pause_mask = oset;
     sigdelset(&pause_mask, SIGCHLD);
+    sigaddset(&pause_mask, SIGINT);
     for (;;) {
 	(void)handle_pending_signals();
 	jobflags = 0;
-- 
2.5.5

